src/pages/signin/LoginForm.js

```diff
@@ -45,16 +48,34 @@ const defaultProps = {
 class LoginForm extends React.Component {
     constructor(props) {
         super(props);
-
         this.onTextInput = this.onTextInput.bind(this);
         this.validateAndSubmitForm = this.validateAndSubmitForm.bind(this);
-
+        this.clearLogin = this.clearLogin.bind(this);
         this.state = {
             formError: false,
             login: '',
         };
     }
 
+    componentDidMount() {
+        if (!canFocusInputOnScreenFocus() || !this.input) {
+            return;
+        }
+        this.input.focus();
+    }
+
+    componentDidUpdate(prevProps) {
+        // Whenever page becomes visible focus the input. We are using withToggleVisibilityView to hide the page instread of unmounting.
```

Comment:
Can probably get rid of this comment. All we need to know is that we focus the input when the component mounts or visibility changes and the code explains it well enough.

Reply:
ok.